### PR TITLE
Update readme with deprecation of operator on newer versions

### PR DESCRIPTION
**Pull request by hectorakemp**
Updates the readme to reflect features introduced in https://access.redhat.com/articles/7028653.

### CODE REVIEW

**Review by mjlshen — APPROVED** [review]
/lgtm

File: README.md [hunk 1]
```diff
@@ -16,7 +16,9 @@ graph LR
   C
   end
 ```
-
+### Deprecation
+On versions of Managed Openshift (OSD/ROSA) greater than version 4.14 (or version 4.13 if the =ext-managed.openshift.io/legacy-ingress-support= flag is switched on for the cluster) the Custom Domains Operator will no longer reconcile new `CustomDomain` objects. Existing `CustomDomain`
+ objects will be coverted to native Openshift `IngressController` resources, and their `HAProxy` workloads rescheduled onto customer worker nodes. Consult https://access.redhat.com/articles/7028653 for further information.
```
Comment (dustman9000) [inline README.md:21]:
```suggestion
 objects will be converted to native Openshift `IngressController` resources, and their `HAProxy` workloads rescheduled onto customer worker nodes. Consult https://access.redhat.com/articles/7028653 for further information.
```

**Comment by jbpratt** [comment]
/cc @ritmun 
/assign

we will need to update the e2e tests to be skipped

**Pull request merged**